Survey/Configurations/UserDefaults/QuestionSubmissionStorage.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  QuestionSubmissionStorage.swift
+//  Survey
+//
+//  Created by Khanh on 07/12/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+// swiftlint:disable let_var_whitespace
+
+protocol QuestionSubmissionCachable {
+
+    var questionsSubmission: [QuestionSubmission] { get set }
+}
+
+struct QuestionSubmissionStorage: QuestionSubmissionCachable {
+
+    @Storage(key: Constants.UserDefaultKeys.questionsSubmission, defaultValue: [])
```

Comment:
I think you can make this contain a static let `shared` that is used in all Impl
```
struct QuestionSubmissionStorage: QuestionSubmissionCachable {
static let shared = QuestionSubmissionStorage()
var questionsSubmission = [QuestionSubmission]()
}
// on VM
questionSubmissionCachable: QuestionSubmissionStorage.shared
```
This way you can make sure it's cleared on every run.

Reply:
It seems not to be working without @Storage, I fixed it at https://github.com/nkhanh44/ic-survey-ios/pull/85/commits/7d9f348d0b973e8751f4571036b45ef196b7f982. Thanks :D.

Reply:
It should be a class because struct cannot change it's property. Adding `shared` this way does not solve the issue we are talking about. 🙇

Reply:
@blyscuit Ohh my bad 🙈 , It's gonna be fixed in PART 2.